### PR TITLE
ADCM-2929 - Implement an adaptive popup

### PR DESCRIPTION
**Pull request by dbardin90**
https://tracker.yandex.ru/ADCM-2929

### CODE REVIEW

File: web/src/app/shared/components/upgrades/master/master.component.ts [hunk 1]
```diff
@@ -87,4 +70,8 @@ export class UpgradeMasterComponent extends BaseDirective implements DynamicComp
   cancel() {
     this.event.emit({ name: 'cancel' });
   }
+
+  needMargin() {
+    return !(this?.model?.upgrades[0]?.config?.config.length > 0 && this?.model?.upgrades[0]?.hostcomponentmap.length > 0);
```
Comment (remizov-arena) [inline web/src/app/shared/components/upgrades/master/master.component.ts:75]: angular test crashed with `TypeError: Cannot read properties of undefined (reading 'length')`
I think we can add `.config?.length` and `hostcomponentmap?.length`

**Review by remizov-arena — APPROVED** [review]

**Comment by kuhella** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @kuhella! Build has been launched!

**Comment by kuhella** [comment]
bender build

**Comment by BenderArenadata** [comment]
Hey, @kuhella! Build has been launched!

**Pull request merged**